shared/js/trackers.js

```diff
@@ -64,13 +66,15 @@ require.scopes.trackers = (function() {
             // tracker is related to the current site. If this is the case we consider it to be the
             // same as a first party requrest and return
             var trackerByParentCompany = checkTrackersWithParentCompany(blockSettings, urlSplit, currLocation)
+            var surrogateTracker = checkSurrogateList(urlToCheck)
```

Comment:
I'd only do this after we have decided to block the request. Look up the surrogate and add the b64 to the data returned from isTracker, then in background.js make the cancel vs redirect to surrogate decision. So `isTracker` might return something like:
```
{ 
    parentCompany: "Google", 
    url: "google-analytics.com", 
    type: "Analytics", 
    block: true 
    surrogate: XXXXXXX
}
```

Reply:
@jdorweiler  are you sure? That requires those to be in the list and certain to block. I see these as separate. In other words, this implementation can handle many other cases that won't be in the block list. This should be super fast lookup.

Reply:
Yup, was about to ask a similar/relevant question: because it is possible to serve up GA from an endpoint that we don't recognize or even from local `file://` protocol, what's to stop someone from evading our surrogate by hosting the file(s) elsewhere?

https://support.google.com/analytics/answer/1032389?hl=en

Reply:
Good point. That sounds fine then. We're still counting them as trackers, right?